blip-tap.md

```diff
@@ -1091,72 +1194,106 @@ receiving node:
 
 - MUST extend a TAP HTLC with an `asset_id` corresponding to the accepted
-  `tap_rfq_scid` with an asset value of: `((amt_asset_incoming *
-  accepted_rate_tick) // msat_multiplier) / tick`
-
-  * As an example:
-    * If the outgoing channel (last hop receiver) is a USD backed asset, and
-      requested an invoice `1_633_054_326 mSAT`, but wants USD with an accepted
-      tick rate of `612_349_500` ($61,234.95), then the penultimate node should
-      send $1000 to the last hop:
-        * `amt_to_forward = ((incoming_amt_msat * accepted_rate_tick) // msat_multiplier) / tick`
-        * `amt_to_forward = ((1_633_054_326 * 612_349_500) // 100_000_000_000) / 10_000`
-        * `amt_to_forward = ((999999999998937000) // 100_000_000_000) / 10_000`
-        * `amt_to_forward = (9999999 / 10_000)`
-        * `amt_to_forward = (9999999 / 10_000)`
-        * `amt_to_forward = 999.9`
-    * Note that all assets internally are accounted in a unit of a `tick`
-      (1/1000th) of an asset. When convering back to the main asset, the value
-      should be rounded up, giving us the original value of `$1000`.
+  `tap_rfq_scid` with an asset value of:
+  ```
+  amt_asset_to_forward = incoming_amt_msat * (asset_to_btc_rate / btc_to_msat_multiplier)
+  ```
+
+  where:
+
+  * `amt_asset_to_forward` is the amount of the TAP asset that the receiver
+    should forward to the next hop.
+
+  * `incoming_amt_msat` is the amount of the incoming HTLC in milli-satoshis.
+
+  * `asset_to_btc_rate` refers to the asset-to-BTC conversion rate established
+    during the RFQ negotiation between the sender and the liquidity provider.
+    This fractional rate is derived from the
+    [fixed-point](#fixed-point-number-type) representation of the conversion
+    rate found in the RFQ wire messages.
 
+  * `btc_to_msat_multiplier` is a constant used to convert BTC to milli-satoshis
+    (msats). It is defined as:
+    ```
+    btc_to_msat_multiplier = 100_000_000 * 1000 = 100_000_000_000`
+    ```
 
+    This constant combines two conversions:
+      1. BTC to Satoshis: 1 BTC = 100,000,000 satoshis.
+      2. Satoshis to msats: 1 satoshi = 1,000 msats.
+
+  Here's an example of how the `amt_asset_to_forward` value is calculated:
+
+  Consider an outgoing channel which uses a USD-backed asset. The last hop
+  receiver issues an invoice for `1_633_054_326 mSAT` but expects USD. The
+  accepted asset-to-BTC conversion rate expressed as a
+  [fixed-point](#fixed-point-number-type) has a coefficient of `612_349_500` and
+  a scale of `4`. The fractional conversion rate is therefore `61_234.95`
+  (implying `$61,234.95 = 1BTC`). As a result, the penultimate node should send
+  $1,000 to the last hop:
+
+  * `amt_asset_to_forward = incoming_amt_msat * (asset_to_btc_rate / btc_to_msat_multiplier)`
+  * `amt_asset_to_forward = 1_633_054_326 * (61_234.95 / 100_000_000_000)`
+  * `amt_asset_to_forward = 999.9999999989369`
+  * `[amt_asset_to_forward] = 1000`
```

Comment:
Actually, I think we only do integer division, so the result would be `999` here. But the invoice would be accepted since we allow up to 1 unit per MPP shard of under-payment.